### ISSUE TITLE
File downloading permission error

### ISSUE THREAD

**Issue by kkiermasz** [Moya/Moya#2102]
Hi,

I want to download and save the file in the `Downloads` directory. File is being downloaded correctly, response code is 200. Unfortunately, Moya gives me `downloadedFileMoveFailed` error. 
In case you need to know, I do not use App Sandbox. You can find how I create destination URL and error description below.
What do I have to do to satisfy Moya permission requirements?

Destination URL (return with URL(safeString:) initializer is never called):
```
 private func makeFileAnchor(for deviceType: String, firmwareVersion: String) -> URL {
        let filename = String("firmware-\(deviceType)-\(firmwareVersion)".hashValue) + ".zip"
        guard let directoryPath = fileManager.urls(for: .downloadsDirectory, in: .userDomainMask).first else {
            return URL(safeString: NSHomeDirectory() + "/Downloads/" + filename)
        }
        return directoryPath.appendingPathComponent(filename)
    }
```

Error:
```
MoyaError
  ▿ underlying : 2 elements
    ▿ .0 : AFError
      ▿ downloadedFileMoveFailed : 3 elements
        - error : Error Domain=NSCocoaErrorDomain Code=513 "You don’t have permission to save the file “Downloads” in the folder “1A6FA1D3-1022-41B3-9F42-696DE3F24A68”." UserInfo={NSFilePath=/var/mobile/Containers/Data/Application/1A6FA1D3-1022-41B3-9F42-696DE3F24A68/Downloads, NSUnderlyingError=0x283419740 {Error Domain=NSPOSIXErrorDomain Code=1 "Operation not permitted"}}
        ▿ source : file:///private/var/mobile/Containers/Data/Application/1A6FA1D3-1022-41B3-9F42-696DE3F24A68/tmp/CFNetworkDownload_5Gp23k.tmp
          - _url : file:///private/var/mobile/Containers/Data/Application/1A6FA1D3-1022-41B3-9F42-696DE3F24A68/tmp/CFNetworkDownload_5Gp23k.tmp
        ▿ destination : file:///var/mobile/Containers/Data/Application/1A6FA1D3-1022-41B3-9F42-696DE3F24A68/Downloads/-1114601813549610444.zip
          - _url : file:///var/mobile/Containers/Data/Application/1A6FA1D3-1022-41B3-9F42-696DE3F24A68/Downloads/-1114601813549610444.zip
    ▿ .1 : Optional<Response>
      ▿ some : Status Code: 200, Data Length: 0
```

Moya 4.0.0
Xcode 12.0
Swift 5.3
iOS 14.0.1

**Comment by sunshinejr** [comment]
@kkiermasz could you try replacing `NSHomeDirectory()` with `directoryPath`?

**Comment by kkiermasz** [comment]
Sure, but I have indicated that this line is never used

**Comment by sunshinejr** [comment]
@kkiermasz yes I saw that, but the error still indicates that it couldn't save file `Downloads`, which I only see in the else in the guard. The default implementation would return documents directory + filename, but it's not what the error suggest: 
`file:///var/mobile/Containers/Data/Application/1A6FA1D3-1022-41B3-9F42-696DE3F24A68/Downloads/-1114601813549610444.zip`

It also doesn't have the `firmware` and `deviceType` in the name and I'm wondering, is there another function that has a similar logic in your app?

**Comment by kkiermasz** [comment]
Why should I expect to get documents directory, like you said, if im asking fileManager for `.downloadsDirectory`? I think that's why error indicated error associated with the `Downloads`.

The name doesn't contain `firmware` and `deviceType` because it is being hashed.

I will try to create this directory first and get back to you with the result, but tbh I expected that `DownloadDestination.Options.createIntermediateDirectories` will also create required directories.

Btw, thank you very much for your quick help!

**Comment by kkiermasz** [comment]
All right, so I have found an easy solution. According to suggestion you quietly smuggled in I have asked the `fileManger` for `.documentDirectory`, not the `.downloadsDirectory`. For now, it satisfies me :D
Btw it would be impossible to create a `Downloads` folder as it already existed before: P

Thank you for help once again!

**Comment by sunshinejr** [comment]
@kkiermasz oh damn, I'm sorry! I'm so used to saving files in `documentsDirectory` that I didn't notice you actually asked for `downloadsDirectory`! 🙈  

Not sure about this option, since I never used `downloadsDirectory` on iOS, but the last time I investigated that the folder wasn't available by default and to store any user-related content you'd use the `documentsDirectory`. I checked the app folder a sec ago and the `Downloads` folder is still not there, but the interesting thing is the `downloadsDirectory` URL is available for iOS, so maybe there is a way to allow you to save in that folder? Not sure, but if you end up finding the answer, be sure to let me know!
